elementary/monitor/data_monitoring/schema.py

```diff
@@ -14,46 +14,88 @@ class InvalidSelectorError(Exception):
     pass
 
 
-class Status(Enum):
+class Status(str, Enum):
     WARN = "warn"
     FAIL = "fail"
     SKIPPED = "skipped"
     ERROR = "error"
     RUNTIME_ERROR = "runtime error"
 
 
-class ResourceType(Enum):
+class ResourceType(str, Enum):
     TEST = "test"
     MODEL = "model"
     SOURCE_FRESHNESS = "source_freshness"
 
 
-class SupportedFilterTypes(Enum):
+class FilterType(str, Enum):
     IS = "is"
+    IS_NOT = "is_not"
+    CONTAINS = "contains"
 
 
-class FilterSchema(BaseModel):
+def apply_filter(filter_type: FilterType, value: Any, filter_value: Any) -> bool:
+    if filter_type == FilterType.IS:
+        return value == filter_value
+    elif filter_type == FilterType.IS_NOT:
+        return value != filter_value
+    elif filter_type == FilterType.CONTAINS:
+        return str(filter_value).lower() in str(value).lower()
+    raise ValueError(f"Unsupported filter type: {filter_type}")
+
+
+ValueT = TypeVar("ValueT")
+
+
+ANY_OPERATORS = [FilterType.IS, FilterType.CONTAINS]
+ALL_OPERATORS = [FilterType.IS_NOT]
+
+
+class FilterSchema(BaseModel, Generic[ValueT]):
     # The relation between values is OR.
-    values: List[Any]
-    type: SupportedFilterTypes = SupportedFilterTypes.IS
+    values: List[ValueT]
+    type: FilterType = FilterType.IS
 
     class Config:
         # Make sure that serializing Enum return values
         use_enum_values = True
 
+    def _apply_filter_type(self, value: ValueT, filter_value: ValueT) -> bool:
+        return apply_filter(self.type, value, filter_value)
+
+    def apply_filter_on_value(self, value: ValueT) -> bool:
+        if self.type in ANY_OPERATORS:
+            return any(
+                self._apply_filter_type(value, filter_value)
+                for filter_value in self.values
+            )
+        elif self.type in ALL_OPERATORS:
+            return all(
+                self._apply_filter_type(value, filter_value)
+                for filter_value in self.values
+            )
+        raise ValueError(f"Unsupported filter type: {self.type}")
+
+    def apply_filter_on_values(self, values: List[ValueT]) -> bool:
+        if self.type in ANY_OPERATORS:
+            return any(self.apply_filter_on_value(value) for value in values)
+        elif self.type in ALL_OPERATORS:
+            return all(self.apply_filter_on_value(value) for value in values)
+        raise ValueError(f"Unsupported filter type: {self.type}")
+
 
-class StatusFilterSchema(FilterSchema):
+class StatusFilterSchema(FilterSchema[Status]):
     values: List[Status]
```

Comment:
you can remove the re-definition of `values` here

Reply:
I thought so too, but apparently not.
https://github.com/pydantic/pydantic/issues/4171